packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2573,7 +2573,11 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     // Allow type annotations inside of a parameter list.
     parseAssignableListItemTypes(param: N.Pattern) {
       if (this.eat(tt.question)) {
-        if (param.type !== "Identifier" && !this.state.isDeclareContext) {
+        if (
+          param.type !== "Identifier" &&
+          !this.state.isDeclareContext &&
```

Comment:
Does `this.state.isDeclareContext` imply `this.state.inType`? If so we can remove `!this.state.isDeclareContext` check.

Reply:
Let me dig deeper, there was some failures when I replaced it originally.

Reply:
Ah I see. `this.state.inType` is technically `this.state.inTypeAnnotation`, so `this.state.isDeclareContext` does not imply `this.state.inType`.

Reply:
Yeah, naming is hard x2 😸